recipe/patch_yaml_utils.py

```diff
@@ -385,7 +385,21 @@ def _pin_looser(fn, record, fix_dep, max_pin=None, upper_bound=None):
         lower = m.group("lower")
         upper = m.group("upper").split(".")
 
-        if upper_bound is None:
+        if upper_bound == "None":
+            msg = (
+                "To remove an upper bound, use `upper_bound: null`; "
+                "`upper_bound: None` wrongly gets interpreted as a string!"
+            )
+            raise ValueError(msg)
+
+        if upper_bound is None and max_pin is None:
```

Comment:
```suggestion
        if (upper_bound is None or str(upper_bound).lower() == "none") and max_pin is None:
```

Reply:
Allowing `"None"` is one thing, but all upper/lower case variants (e.g. `"nOnE"`) is just gratuitous divergence and we shouldn't saddle ourselves with supporting that.

Reply:
It's not that bad but I'll back this out. This none is a drop in the bucket.